Fixing mac-pip builds

caisq opened the pull request:
The --system-site-package option flag in PR 1472 broke Mac pip builds due to existing packages on the system. For example, see: http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-mac-python2-copt_both/17/console

Generally speaking, the --system-site-package option makes the virtualenv environment depend on the packages on the Mac host. Removing the option gets rid of this dependency and the need to maintain the host packages, which will save work (especially if there are multiple Mac slaves). 

The cost of installing the packages in the virtualenv should be minimal for both Mac and Linux, now that we cache the virtualenv folder by default. 

The PR has been tested manually for both python2 and python3 on Mac. See:
http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-mac-python2-copt_both/18/console
http://ci.tensorflow.org/view/Experimental/job/experimental-cais-tensorflow-mac-python3-copt_both/4/console

jendap commented:
And how is "blaze test" without pip.sh going to work?

caisq commented:
@jendap For Mac, "blaze test" should work as long as the correct versions of dependent python packages (numpy, scipy and sklearn) are installed on he system. I think the main problem with "--system-site-packages" is that if TensorFlow is already installed on the system, "pip install" in virtualenv from pip.sh will not update it, if that the version number has not increased. Even the "--force-reinstall" option does not work.

jendap commented:
It would be great to have consistency with what bazel see and what virtualenv see.

Looking at the output from your build 17 - it says "Requirement already satisfied (use --upgrade to upgrade): tensorflow==0.7.1". Have you tried adding "--upgrade" to pip install?

Ideally it should upgrade tensorflow but not the other packages.

jendap commented:
This is what we need: `pip install --upgrade --no-deps --force-reinstall <packagename>`

(second answer on http://stackoverflow.com/questions/19548957/can-i-force-pip-to-reinstall-the-current-version)

jendap commented:
Take a look at https://github.com/tensorflow/tensorflow/pull/1495. It does exactly what we need.

caisq commented:
Added the --upgrade suggested in #1495. Ready to merge.

caisq committed:
Fixing mac-pip builds

Using --upgrade flag to make sure that the latest pip whl built from source gets installed. This addresses the Mac pip build issue.

caisq commented:
@tensorflow-jenkins test this please

martinwicke commented:
I merged #1495 so I'm closing this.
